Add a no-op Logger

filiphr opened the pull request:
We have recently moved to Java 17 and are starting to use Microhttp as a mock server (we need to validate our requests).

Having a no-op Logger makes it easier to disable the logging.

ebarlas commented:
Yes, I reach for this type of thing often.

Would you replace `DisabledLogger` in `EventLoopConcurrencyTest` as part of this change?

Also, I'm a little wary of adding the `noop()` method to the `Logger` interface. I think I'd prefer to keep that interface surface area to an absolute minimum.

filiphr commented:
>Would you replace `DisabledLogger` in `EventLoopConcurrencyTest` as part of this change?

Done.

>Also, I'm a little wary of adding the `noop()` method to the `Logger` interface. I think I'd prefer to keep that interface surface area to an absolute minimum.

Sure thing. Removed `noop()`, made `NoopLogger` public and added an `instance()` method there to get an instance of it.

I pushed a new commit so that it is easier to review. However, please feel free to squash them when merging.

>Also, what project are you using this in out of curiosity?

Currently I am using it for some testing in our enterprise Flowable product. Our Open Source project is https://github.com/flowable/flowable-engine. However, we are not using it there yet. We just moved to Java 17 and I can start exploring it more. In the enterprise we are using [MockWebServer](https://github.com/square/okhttp/tree/master/mockwebserver) from OkHttp and I was OK with their version 3. However, they have moved to using Kotlin and thus pulling some Kotlin dependencies so we have not moved there to a newer version.

My goal is to somehow create a small wrapper on top of Microhttp that can be used for testing. Providing easy way to create responses, have some default handler and things like that. I know that this project is purely for the low level bits. However, if you are interested in having something like `microhttp-test` which easier way for people to use Microhttp in a testing scenario I am more than happy to do such a contribution to this project. Another alternative would also be to provide an integration as part of [JUnit Pioneer](https://github.com/junit-pioneer/junit-pioneer) (in case the team there is interested), if none of that works I will see what the best way to achieve this would be

ebarlas commented:
Looks good, thanks for the changes.

And thanks for the usage context. I'm very glad to hear that you're getting mileage out of the library.
